src/map/alghom.jl

```diff
@@ -0,0 +1,142 @@
+export Algebra_Homomorphism, codomain, compose, domain, kernel, preimage
+
+###############################################################################
+#
+#   I/O for Algebra Homomorphisms
+#
+###############################################################################
+
+function show(io::IO, M::Singular.Map(SAlgHom))
+   println(io, "Algebra Homomorphism with")
+   println(io, "")
+   println(io, "Domain: ", domain(M))
+   println(io, "")
+   println(io, "Codomain: ", codomain(M))
+   println(io, "")
+   println(io, "Defining Equations: ", gens(M.image))
+end
+
+###############################################################################
+#
+#   Basic Operations with Algebra Homomorphisms
+#
+###############################################################################
+
+function map_ideal(f::AbstractAlgebra.Map(SAlgHom), I::sideal)
+
+   if base_ring(I) != f.domain
+      error("Ideal is not in the domain of the
+            algebra homomorphism.")
+   end
+
+   return Ideal(f.codomain, libSingular.maMapIdeal(I.ptr, f.domain.ptr,
+                f.image.ptr, f.codomain.ptr))
+end
+
+function map_poly(f::AbstractAlgebra.Map(SAlgHom), p::spoly)
+
+   if parent(p) != f.domain
+      error("Polynomial is not in the domain of the
+            algebra homomorphism.")
+   end
+   return f.codomain(libSingular.maMapPoly(p.ptr, f.domain.ptr, f.image.ptr,
+          f.codomain.ptr))
+end
+
+function (f::SAlgHom)(p::spoly)
+   return map_poly(f, p)
+end
+
+function (f::SAlgHom)(I::sideal)
+   return map_ideal(f, I)
+end
+
+###############################################################################
+#
+#   Composition Algebra Homomorphisms
```

Comment:
Maybe "Composition of Algebra Homomorphisms" is better, to avoid any confusion.

Reply:
Done.